### PR TITLE
DT-753: Replace GitHub Actions access keys with OIDC

### PR DESCRIPTION
**Pull request by YD-DLUHC**
This PR creates the roles and policies required to use OIDC instead of secrets/keys within GitHub Actions.

### CODE REVIEW

**Review by hugh-emerson — APPROVED** [review]
LGTM

File: terraform/production/ci_iam.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      "ecr:CompleteLayerUpload",
+      "ecr:UploadLayerPart",
+      "ecr:DescribeImages",
+      "ecr:InitiateLayerUpload",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:PutImage"
+    ]
+
+    resources = [
+      "468442790030.dkr.ecr.eu-west-1.amazonaws.com/delta-auth-service"
```
[the remark this answers is not in the document]
Comment (YD-DLUHC) [reply terraform/production/ci_iam.tf]: Have changed it to use data.aws_caller_identity.current.account_id if that's what you meant?

File: terraform/production/ci_iam.tf [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+
+    condition {
+      test = "StringLike"
+      values = [
+        "repo:communitiesuk/delta-auth-service:*"
```
Comment (BenRamchandani) [inline terraform/production/ci_iam.tf:31]: This should be restricted to the "publish" environment

File: terraform/production/ci_iam.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenRamchandani) [reply terraform/production/ci_iam.tf]: That's probably fine. You could make a local var like Hugh suggested

```hcl
locals {
  ecr_repo = "${data.aws_caller_identity.current.account_id}.dkr.ecr.eu-west-1.amazonaws.com/delta-auth-service"
}
```
and then reference it here and where it's used in main.tf

**Review by BenRamchandani — CHANGES_REQUESTED** [review]

File: terraform/production/ci_iam.tf (hunk 2 above)
Comment (YD-DLUHC) [reply terraform/production/ci_iam.tf]: changed to restrict to publish

File: terraform/production/ci_iam.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    condition {
+      test     = "StringEquals"
+      values   = ["sts.amazonaws.com"]
+      variable = "token.actions.githubusercontent.com:aud"
+    }
+
+    condition {
+      test = "StringLike"
```
Comment (BenRamchandani) [inline terraform/production/ci_iam.tf:33]:
```suggestion
      test = "StringEquals"
```

**Review by BenRamchandani — APPROVED** [review]

**Pull request merged**